c++/src/kj/encoding-test.c++

```diff
@@ -241,14 +241,26 @@ KJ_TEST("round-trip invalid UTF-16") {
 }
 
 KJ_TEST("EncodingResult as a Maybe") {
-  KJ_IF_MAYBE(result, encodeUtf16("\x80")) {
-    KJ_FAIL_EXPECT("expected failure");
+  {
+    auto result = encodeUtf16("\x80");
+    KJ_EXPECT(result != nullptr);  // It has bytes ...
+    KJ_EXPECT(result == kj::none); // But an error.
+    KJ_IF_SOME(unused, result) {
+      (void)unused;
```

Comment:
Hmm, does KJ_IF_SOME generate unused variable warnings where KJ_IF_MAYBE didn't? Can we inhibit this with KJ_UNUSED? Seems annoying.

Reply:
We can, but I see it as a benefit, myself. So, I'll merge as-is, but if it gets annoying in practice, I don't mind if we change it.